### PR TITLE
Fix bug where an enemy army could retreat immediately when loading a save.

### PR DESCRIPTION
**Pull request by PrettyFlower**

### CODE REVIEW

File: src/figure/enemy_army.c [hunk 1]
```diff
@@ -189,6 +189,9 @@ void enemy_armies_load_state(buffer *buf, buffer *totals_buf)
     for (int i = 0; i < MAX_ENEMY_ARMIES; i++) {
         enemy_armies[i].ignore_roman_soldiers = buffer_read_i32(buf);
     }
+    for (int i = 0; i < MAX_ENEMY_ARMIES; i++) {
+        enemy_armies[i].started_retreating = 0;
```
[the remark this answers is not in the document]
Comment (PrettyFlower) [reply src/figure/enemy_army.c:193]: Yeah, I did it separately so it would look like all the other assignments in that function.

**Comment by PrettyFlower** [comment]
I’ll need to check later but it might still be possible to get this bug if the player starts a city from the city construction kit after causing a retreat in a different city. Right now I’m only clearing the started_retreating flag on load but there may be other places we need to unset it.

**Comment by crudelios** [comment]
Okay then I'll keep this open for now 👍

**Comment by PrettyFlower** [comment]
Ok after extensive checking, I believe this works in the following 3 cases:
1. When loading an existing saved game, the flag is cleared with the new line added to `enemy_armies_load_state()`.
2. When loading a custom scenario (i.e. city construction kit), the flag is cleared in `enemy_armies_clear()`.
3. When loading a regular scenario, `enemy_armies_clear()` is not called but apparently `enemy_armies_load_state()` is, so I guess regular scenarios are just loading saved games.

File: src/figure/enemy_army.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (PrettyFlower) [reply src/figure/enemy_army.c:191]: I've gone ahead and moved this up per your comment, but missed the one below where you said you were going to keep this open until I had checked the other cases. Let me know if I can put it back in its own loop for consistency.

[the remark this answers is not in the document]
Comment (crudelios) [reply src/figure/enemy_army.c:191]: I see what you mean.

I don't know why the savegame was created like this, requiring a huge number of `for` loops.

Almost wants to make me suggest bumping the savegame version and using a single `for` loop for new cases. Even better, reducing the size to the actual number of existing armies. Maybe after v3.2.0 is released.

Right now, I'll leave it to you. If you'd prefer a separate `for` loop for zeroing the `started_retreating` I won't nag 😉

[the remark this answers is not in the document]
Comment (PrettyFlower) [reply src/figure/enemy_army.c:193]: Okie dokie, I put it back. Thx!

**Pull request merged**